src/destructors.md

```diff
@@ -63,10 +63,11 @@ loop {
     moved = ShowOnDrop("Drops when moved");
     // drops now, but is then uninitialized
     moved;
+    // Uninitialized does not drop.
     let uninitialized: ShowOnDrop;
     // Only first element drops
-    let mut partially_initialized: (ShowOnDrop, ShowOnDrop);
-    partially_initialized.0 = ShowOnDrop("Partial tuple first");
+    let mut partially_initialized = (ShowOnDrop("one"), ShowOnDrop("two"));
+    core::mem::forget(partially_initialized.1);
```

Comment:
Just above there is a move example that actually drops early by moving, so maybe the same would make sense here?
```
// drops now, but second field (.1) is then uninitialized
partially_initialized.1;
// first field (.0) implicitly dropped.
```

Reply:
Yeah that's probably better stylistically.

Reply:
I like the explicitness of calling `forget`, since if you run the example you can actually see how it only drops the remaining fields.  What about the following text to be more explicit (and to match the wording in rustc):

```rust
    // After a partial move, only the remaining fields are dropped.
    let mut partial_move = (ShowOnDrop("first"), ShowOnDrop("forgotten"));
    // Perform a partial move, leaving `partial_move.0` initialized.
    core::mem::forget(partial_move.1);
    // When `partial_move` scope ends, only the first field is dropped.
```

Reply:
Some wording nits: "leaving" -> "leaving only"

"`partial_move`" -> "`partial_move`'s"

Aside from that I think it is good.
Imo we can land this and iterate on style :)